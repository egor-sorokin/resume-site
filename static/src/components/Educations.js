import React, { Component } from 'react';
import withFetching from '../utils/api';
import { URL_PATH_EDUCATION } from '../constants/index';


const Educations = ({ data, isFetching, error }) => {
  const educations = data.educations || [];

  if (error) {
    return <p>{error.message}</p>;
  }

  if (isFetching) {
    return <p>Loading ...</p>;
  }

  return (
    <div>
      dasd
      <div>
        {educations.map(item =>
          <div key={item.id}>
            asd
          </div>
        )}
      </div>
    </div>
  );
};

export default withFetching(URL_PATH_EDUCATION)(Educations);