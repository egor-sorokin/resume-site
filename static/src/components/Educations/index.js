import React from 'react';
import PropTypes from 'prop-types';
import withFetching from '../../utils/api';
import { URL_PATH_EDUCATIONS } from '../../constants/index';


function Educations({ data, isFetching, error }) {
  const educations = data.educations || [];
  const educationComponent = educations.map(item => (
    <div key={item.id}>
      asd
    </div>
  ));

  if (error) {
    return <div><p>{error}</p></div>;
  }

  if (isFetching) {
    return <div><p>Loading...</p></div>;
  }

  return (
    <div>
      {educationComponent}
    </div>
  );
}

Educations.propTypes = {
  data: PropTypes.shape([
    PropTypes.shape({
      id: PropTypes.number,
    }),
  ]),
  isFetching: PropTypes.bool,
  error: PropTypes.string,
};


Educations.defaultProps = {
  data: [],
  error: '',
  isFetching: false,
};


export default withFetching(URL_PATH_EDUCATIONS, Educations);
