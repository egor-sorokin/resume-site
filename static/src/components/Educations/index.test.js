import React from 'react';
import Educations from './index';
import { shallow, render, mount } from 'enzyme';

describe('<Educations />', () => {
  it('renders 1 <Educations /> component', () => {
    const component = shallow(<Educations />);
    expect(component).toHaveLength(1);
  })
});
