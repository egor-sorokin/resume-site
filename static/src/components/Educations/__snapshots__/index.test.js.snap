// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Educations /> matches snapshot 1`] = `
<Educations
  data={Object {}}
  error={null}
  isFetching={false}
/>
`;
